File: app/src/main/java/com/agrogames/islandsofwar/engine/abs/renderable/RenderableObject.java
package com.agrogames.islandsofwar.engine.abs.renderable;

import com.agrogames.islandsofwar.engine.abs.common.Point;

public interface RenderableObject {
    Point getLocation();
    float getRotation();
    String getTexture();
    default float timeSinceDestroyed(){
        return 0;
    }
}

File: app/src/main/java/com/agrogames/islandsofwar/engine/abs/renderable/RenderableObject.kt
package com.agrogames.islandsofwar.engine.abs.renderable

import com.agrogames.islandsofwar.engine.abs.common.Point

interface RenderableObject {
    val location: Point
    val rotation: Float
    val texture: String
    fun timeSinceDestroyed(): Float {
        return 0f
    }
}